src/python/WMCore/WMSpec/Steps/Executors/DQMUpload.py

```diff
@@ -244,7 +245,7 @@ def upload(self, url, args, filename):
         opener.add_handler(handler)
 
         # setup the request object
-        datareq = urllib2.Request(url + '/data/put')
+        datareq = urllib2.Request(encodeUnicodeToBytes(url) + '/data/put')
```

Comment:
comment not related to the review, just something to ponder about tonight. Do we want something similar in master? something like
```suggestion
        if PY2:
            url = encodeUnicodeToBytes(url)
        elif PY3:
            url = decodeBytesToUnicode(url)
        datareq = urllib2.Request(url + '/data/put')
```

Reply:
That's a good idea. Can you please remind me whether we still have to re-apply the future layer to have this module compatibility with both python2 and python3? Or is it already dual stack?

Reply:
Aha! you are correct, `src/python/WMCore/WMSpec/Steps/Executors/DQMUpload.py ` is definitely not py3 compatible, so we can just add this to the list of changes that it needs to be py3 compatible

Reply:
Thank you very much for the review and for getting this fixed!